Support CI - Please review a bit before merge

iongion opened the pull request:
Hi,

This PR only deals with the testing aparatus and not the tests themselves.
If you have a travis-ci account, you should import this project there, I do not have enough permissions.

![image](https://user-images.githubusercontent.com/867609/43847570-32cb57d0-9b31-11e8-9648-0239fc297ffc.png)

Tests pass (xvfb)!

iongion committed:
properly install lua and run tests

iongion committed:
Added build status badge for master

iongion committed:
ci support (badge, deps, build, test)

iongion committed:
build on travis

starwing commented:
hi, for travis-ci, I advised to use hererocks to make up lua environment, you can see my other projects for that: https://github.com/starwing/lua-protobuf/blob/master/.travis.yml#L22

iongion commented:
@starwing Yes, I will do that, I was losing my mind setting up an environment that works for my machine but also for Travis

iongion committed:
remove unused scripts
fix ignore
initial support for virtualenv
avoid downloading and building libraries

iongion committed:
ensure moonglfw is loaded before nvg

iongion committed:
added demo phase

iongion committed:
proper demo sourcing

iongion committed:
goal update

starwing commented:
I have actived the travis-ci of this project.  maybe you could just push another commit to see whether it works :)

iongion committed:
speed up test goal
add test from README
remove unused prints
fix undefined symbol error due to moonglfw not linking against proper lua

starwing commented:
it seems that your script install moonglfw failed:

```shell
$ bash ./scripts/build.sh
Setting up runtime configuration
Detected PLATFORM=linux machine=Linux extension=so lua=5.3
Building in /home/travis/build/starwing/lua-nanovg/build/linux
Ensuring build directory in /home/travis/build/starwing/lua-nanovg/build/linux
Building project starting
Cleaned build files
make[1]: Entering directory '/home/travis/build/starwing/lua-nanovg/moonglfw'
make[2]: Entering directory '/home/travis/build/starwing/lua-nanovg/moonglfw/src'
make[2]: Leaving directory '/home/travis/build/starwing/lua-nanovg/moonglfw/src'
make[2]: Entering directory '/home/travis/build/starwing/lua-nanovg/moonglfw/doc'
make[2]: Leaving directory '/home/travis/build/starwing/lua-nanovg/moonglfw/doc'
make[1]: Leaving directory '/home/travis/build/starwing/lua-nanovg/moonglfw'
Installing moonglfw dependency in /home/travis/.luarocks/lib/lua/5.3
cp: cannot stat 'moonglfw.so': No such file or directory
Makefile:72: recipe for target 'install' failed
make: *** [install] Error 1
```

iongion commented:
Yes, I am working on it, I will ping you for a new review asap

On Mon, Aug 13, 2018 at 9:01 AM, Xavier Wang <notifications@github.com>
wrote:

> it seems that your script install moonglfw failed:
>
> $ bash ./scripts/build.sh
> Setting up runtime configuration
> Detected PLATFORM=linux machine=Linux extension=so lua=5.3
> Building in /home/travis/build/starwing/lua-nanovg/build/linux
> Ensuring build directory in /home/travis/build/starwing/lua-nanovg/build/linux
> Building project starting
> Cleaned build files
> make[1]: Entering directory '/home/travis/build/starwing/lua-nanovg/moonglfw'
> make[2]: Entering directory '/home/travis/build/starwing/lua-nanovg/moonglfw/src'
> make[2]: Leaving directory '/home/travis/build/starwing/lua-nanovg/moonglfw/src'
> make[2]: Entering directory '/home/travis/build/starwing/lua-nanovg/moonglfw/doc'
> make[2]: Leaving directory '/home/travis/build/starwing/lua-nanovg/moonglfw/doc'
> make[1]: Leaving directory '/home/travis/build/starwing/lua-nanovg/moonglfw'
> Installing moonglfw dependency in /home/travis/.luarocks/lib/lua/5.3
> cp: cannot stat 'moonglfw.so': No such file or directory
> Makefile:72: recipe for target 'install' failed
> make: *** [install] Error 1
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/starwing/lua-nanovg/pull/13#issuecomment-412426002>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA09GeN-vCYUsBbZNzzhQms9alVf0qJ1ks5uQSRSgaJpZM4V0LAN>
> .
>



-- 
*Ionut Stoica <http://www.zend.com/en/yellow-pages/ZEND016175>*
Zend Certified Engineer <http://www.zend.com/en/yellow-pages/ZEND016175>
